### PR TITLE
Fix PowerShell module for release configuration

### PR DESCRIPTION
**Pull request by ryfu-msft**
Changes:
- Fixes a bug where the release configuration build for the PowerShell module was throwing a type initialization error. 
- Updates the build output for Detours and UndockedRegFreeWinRT to the appropriate location
- Added an post-build event to copy the modules files into its own directory so that it can be run directly
- Added a README with instructions on how to run the powershell module locally.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/2599)

### CODE REVIEW

File: src/PowerShell/Microsoft.WinGet.Client/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿# Windows Package Manager PowerShell Module
+
+This project contains the source code for building the Windows Package Manager PowerShell Module.
+
+## Running the PowerShell Module Locally
+
+After building the project solution, the `Microsoft.WinGet.Client` PowerShell module can be found in the output directory in the `PowerShell` folder. For example if you built the project as x64 release, you should expect to find the module files in `$(ProjectDirectory)/src/x64/Release/PowerShell`.
```
Comment (yao-msft) [inline src/PowerShell/Microsoft.WinGet.Client/README.md:7]:
```suggestion
After building the project solution, the `Microsoft.WinGet.Client` PowerShell module can be found in the output directory in the `PowerShell` folder. For example if you built the project as x64 release, you should expect to find the module files in `$(SolutionDirectory)/src/x64/Release/PowerShell`.

```

File: src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/UndockedRegFreeWinRT.vcxproj [hunk 2]
```diff
@@ -73,26 +73,26 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <TargetName>winrtact</TargetName>
     <LinkIncremental>true</LinkIncremental>
-    <OutDir>$(BuildOutDir)</OutDir>
-    <IntDir>$(OutDir)$(ProjectName)</IntDir>
+    <IntDir>$(SolutionDir)$(Platform)\$(Configuration)\$(ProjectName)\</IntDir>
+    <OutDir>$(Platform)\$(Configuration)\</OutDir>
```
Comment (yao-msft) [inline src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/UndockedRegFreeWinRT.vcxproj:77]: Are these opposite? I think we wanted build outputs to go to $(SolutionDir)$(Platform)\$(Configuration)\$(ProjectName)

File: src/PowerShell/Microsoft.WinGet.Client/README.md (hunk 1 above)
Comment (ryfu-msft) [reply src/PowerShell/Microsoft.WinGet.Client/README.md]: Changed to suggested

**Review by yao-msft — APPROVED** [review]

**Pull request merged**